workers/unity/Packages/com.improbable.gdk.mobile/Android/Utility/LaunchArguments.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+
+namespace Improbable.Gdk.Mobile.Android
+{
+    public static class LaunchArguments
+    {
+        public static Dictionary<string, string> GetArguments()
+        {
+            try
+            {
+                using (var unityPlayer = new UnityEngine.AndroidJavaClass("com.unity3d.player.UnityPlayer"))
+                using (var currentActivity = unityPlayer.GetStatic<UnityEngine.AndroidJavaObject>("currentActivity"))
+                using (var intent = currentActivity.Call<UnityEngine.AndroidJavaObject>("getIntent"))
+                {
+                    var hasExtra = intent.Call<bool>("hasExtra", "arguments");
+                    if (hasExtra)
+                    {
+                        using (var extras = intent.Call<UnityEngine.AndroidJavaObject>("getExtras"))
+                        {
+                            var arguments = extras.Call<string>("getString", "arguments").Split(' ');
+                            return CommandLineUtility.ParseCommandLineArgs(arguments);
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogException(e);
```

Comment:
when could an exception be thrown? do we just want to log it or actually throw it?

Reply:
It can be thrown is this is run on an android version that has breaking changes on this api. Or a newer version of unity that breaks here.
Instead of breaking the entire launch button, it just breaks the auto connect.